mllib/src/main/scala/org/apache/spark/ml/classification/NaiveBayes.scala

```diff
@@ -109,10 +118,89 @@ class NaiveBayes @Since("1.5.0") (
         s" numClasses=$numClasses, but thresholds has length ${$(thresholds).length}")
     }
 
-    val oldDataset: RDD[OldLabeledPoint] =
-      extractLabeledPoints(dataset).map(OldLabeledPoint.fromML)
-    val oldModel = OldNaiveBayes.train(oldDataset, $(smoothing), $(modelType))
-    NaiveBayesModel.fromOld(oldModel, this)
+    val numFeatures = dataset.select(col($(featuresCol))).head().getAs[Vector](0).size
+
+    val requireNonnegativeValues: Vector => Unit = (v: Vector) => {
```

Comment:
@zhengruifeng (minor) this method is static and does not depend on the state of the object. It is preferable to put it in the companion object. Also, because people do not have a functional programing background, it is better to define it with `def` rather than a `val`.